### PR TITLE
Update GesturesActivity example to match documentation

### PR DESCRIPTION
**Pull request by ZiZasaurus**
This PR updates the GesturesActivity example to match our documentation on [listening to gesture events](https://docs.mapbox.com/android/maps/guides/user-interaction/#listen-for-gesture-events). Now, gestures are added from the gesturesPlugin and removed within `onDestroy`. This will prevent any confusion for developers when manipulating/detecting gestures within the map view. 

Fixes https://github.com/mapbox/mapbox-maps-android/issues/1641

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/GesturesActivity.kt [hunk 1]
```diff
@@ -71,6 +76,14 @@ class GesturesActivity : AppCompatActivity() {
     gestureAlertsAdapter.cancelUpdates()
   }
 
+  override fun onDestroy() {
+    super.onDestroy()
+    gesturesPlugin.removeOnMoveListener(moveListener)
+    gesturesPlugin.removeOnRotateListener(rotateListener)
+    gesturesPlugin.removeOnScaleListener(scaleListener)
+    gesturesPlugin.removeOnShoveListener(shoveListener)
```
Comment (jush) [inline app/src/main/java/com/mapbox/maps/testapp/examples/GesturesActivity.kt:84]: Nit: It probably doesn't matter but I would change the order just in case `gesturesPlugin` would do something to the activity.
```suggestion
    gesturesPlugin.removeOnMoveListener(moveListener)
    gesturesPlugin.removeOnRotateListener(rotateListener)
    gesturesPlugin.removeOnScaleListener(scaleListener)
    gesturesPlugin.removeOnShoveListener(shoveListener)
    super.onDestroy()
```

File: app/src/main/java/com/mapbox/maps/testapp/examples/GesturesActivity.kt [hunk 2]
```diff
@@ -41,6 +42,10 @@ class GesturesActivity : AppCompatActivity() {
   private var focalPointLatLng: Point? = null
   private var pointAnnotationManager: PointAnnotationManager? = null
   private lateinit var binding: ActivityGesturesBinding
+  private lateinit var rotateListener: OnRotateListener
```
Comment (jush) [inline app/src/main/java/com/mapbox/maps/testapp/examples/GesturesActivity.kt:45]: Actually, I think `gestureAlertsAdapter` doesn't need to be `lateinit var`, it could be `val` and initialized already at definition:
```Kotlin
private val gestureAlertsAdapter = GestureAlertsAdapter()
```
Then, these `*Listener`s could also be just `val` instead of `lateinit var`.

Wdyt? Would it be ok to do that change here or shall we do it in another PR?

Comment (ZiZasaurus) [reply app/src/main/java/com/mapbox/maps/testapp/examples/GesturesActivity.kt:45]: I'm okay with making the change here.

**Review by jush — APPROVED** [review]

File: app/src/main/java/com/mapbox/maps/testapp/examples/GesturesActivity.kt [hunk 3]
```diff
@@ -37,10 +38,93 @@ class GesturesActivity : AppCompatActivity() {
   private lateinit var mapboxMap: MapboxMap
   private lateinit var gesturesPlugin: GesturesPlugin
   private lateinit var gesturesManager: AndroidGesturesManager
-  private lateinit var gestureAlertsAdapter: GestureAlertsAdapter
+  private val gestureAlertsAdapter: GestureAlertsAdapter = GestureAlertsAdapter()
   private var focalPointLatLng: Point? = null
   private var pointAnnotationManager: PointAnnotationManager? = null
   private lateinit var binding: ActivityGesturesBinding
+  private val rotateListener: OnRotateListener = object : OnRotateListener {
+    override fun onRotateBegin(@NonNull detector: RotateGestureDetector) {
```
Comment (kiryldz) [inline app/src/main/java/com/mapbox/maps/testapp/examples/GesturesActivity.kt:46]: `@NonNull` is not needed here and below, it's actually visible from `RotateGestureDetector` type and not `RotateGestureDetector?`

**Review by kiryldz — APPROVED** [review]
Please run `./gradlew ktlint` locally, fix the issues and then good to go!

**Review by pengdev — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]
